### PR TITLE
tests, reporter: If the suite fails, execute the reporter

### PR DESCRIPTION
**Pull request by EdDev**
**What this PR does / why we need it**:

After all tests finished and all teardown fixtures executed, report
the state of the cluster.
    
This is useful for cases where all tests pass but the suite fails, due
to one of its teardown cleanups (e.g. waiting for all test namespaces to
be deleted).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
Error occurred and artifacts collected: https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/10692/pull-kubevirt-e2e-k8s-1.28-sig-storage/1721880113642999808/artifacts/k8s-reporter/suite/

**Comment by akalenyu** [comment]
> Error occurred and artifacts collected: https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/10692/pull-kubevirt-e2e-k8s-1.28-sig-storage/1721880113642999808/artifacts/k8s-reporter/suite/

Yeah looks like we fail to remove snapshot finalizer
```
"finalizers": [
                    "snapshot.kubevirt.io/snapshot-source-protection"
                ],
```

**Comment by mhenriks** [comment]
Awesome!  I will try to figure out why that finalizer could be hanging on.

/lgtm
/approve

**Review by dhiller — COMMENTED** [review]
@EdDev thank you for this, great stuff. One question I have inline though.

/hold

File: tests/tests_suite_test.go [hunk 1]
```diff
@@ -123,12 +123,20 @@ func getMaxFailsFromEnv() int {
 	return maxFails
 }
 
+var _ = ReportAfterSuite("Collect cluster data", func(report Report) {
+	artifactPath := filepath.Join(flags.ArtifactsDir, "k8s-reporter", "suite")
+	kvReport := reporter.NewKubernetesReporter(artifactPath, 1)
+	kvReport.Cleanup()
```
[the remark this answers is not in the document]
Comment (dhiller) [reply tests/tests_suite_test.go:129]: Oh, now I see that you are using a subfolder inside k8s-reporter. Nevermind :)

**Review by dhiller — APPROVED** [review]
/unhold

**Pull request merged**